api/src/services/comments.ts

```diff
@@ -0,0 +1,463 @@
+		this.serviceOrigin = options.serviceOrigin ?? 'comments';
+	}
+
+	override readOne(key: PrimaryKey, query?: Query, opts?: QueryOptions): Promise<Item> {
+		const isLegacyComment = !isNaN(Number(key));
+
+		let result;
+
+		if (isLegacyComment) {
+			const activityQuery = this.serviceOrigin === 'activity' ? query : this.generateQuery('activity', query || {});
+			result = this.activityService.readOne(key, activityQuery, opts);
+		} else {
+			const commentsQuery = this.serviceOrigin === 'comments' ? query : this.generateQuery('comments', query || {});
+			result = super.readOne(key, commentsQuery, opts);
+		}
+
+		return result;
+	}
+
+	override async readByQuery(query: Query, opts?: QueryOptions): Promise<Item[]> {
+		const activityQuery = this.serviceOrigin === 'activity' ? query : this.generateQuery('activity', query);
+		const commentsQuery = this.serviceOrigin === 'comments' ? query : this.generateQuery('comments', query);
+		const activityResult = await this.activityService.readByQuery(activityQuery, opts);
+		const commentsResult = await super.readByQuery(commentsQuery, opts);
+
+		if (query.aggregate) {
+			// Merging the first result only as the app does not utilise group
+			return [
+				mergeWith({}, activityResult[0], commentsResult[0], (a: any, b: any) => {
+					if (isNumber(a) && isNumber(b)) {
+						return a + b;
+					}
+
+					return;
+				}),
+			];
+		} else if (query.sort) {
+			return this.sortLegacyResults([...activityResult, ...commentsResult], query.sort);
+		} else {
+			return [...activityResult, ...commentsResult];
+		}
+	}
+
+	override async readMany(keys: PrimaryKey[], query?: Query, opts?: QueryOptions): Promise<Item[]> {
+		const commentsKeys = [];
+		const activityKeys = [];
+
+		for (const key of keys) {
+			if (isNaN(Number(key))) {
+				commentsKeys.push(key);
+			} else {
+				activityKeys.push(key);
+			}
+		}
+
+		const activityQuery = this.serviceOrigin === 'activity' ? query : this.generateQuery('activity', query || {});
+		const commentsQuery = this.serviceOrigin === 'comments' ? query : this.generateQuery('comments', query || {});
+		const activityResult = await this.activityService.readMany(activityKeys, activityQuery, opts);
+		const commentsResult = await super.readMany(commentsKeys, commentsQuery, opts);
+
+		if (query?.sort) {
+			return this.sortLegacyResults([...activityResult, ...commentsResult], query.sort);
+		} else {
+			return [...activityResult, ...commentsResult];
+		}
+	}
+
+	override async createOne(data: Partial<Comment>, opts?: MutationOptions): Promise<PrimaryKey> {
+		if (!data['comment']) {
+			throw new InvalidPayloadError({ reason: `"comment" is required` });
+		}
+
+		if (!data['collection']) {
+			throw new InvalidPayloadError({ reason: `"collection" is required` });
+		}
+
+		if (!data['item']) {
+			throw new InvalidPayloadError({ reason: `"item" is required` });
+		}
+
+		if (this.accountability) {
+			await validateAccess(
```

Comment:
This throws an `INTERNAL_SERVER_ERROR` response when the collection is non-existent. 

Server side response error is `Cannot find primary key for collection \"fake_collection\""` instead of the expected `You don't have permission to access collection \"article\" or it does not exist. Queried in root`